PS-107 ENH Centreon-HA 22.10 2 nodes constraints mysql-status acceptance guide (#1971)
centreon/centreon-documentation: versioned_docs/version-22.10/administration/centreon-ha/acceptance-guide.md

pcs resource clear centreon
```

Also, check that MySQL replication is still operational using the command:

```bash
/usr/share/centreon-ha/bin/mysql-check-status.sh
```

The command must return the following information:

```text
Connection MASTER Status '@CENTRAL_MASTER_NAME@' [OK]
Connection SLAVE Status '@CENTRAL_SLAVE_NAME@' [OK]
Slave Thread Status [OK]
Position Status [OK]
```

> > If the synchronization has `KO` you have to fix it with the help of the [operating-guide](operating-guide.md).

Reviewer: ```suggestion
> If the synchronization is `KO` you have to fix it with the help of the [operating-guide](operating-guide.md).
```

    * vip       (ocf::heartbeat:IPaddr2):        Started @CENTRAL_SLAVE_NAME@
    * http      (systemd:httpd):         Started @CENTRAL_SLAVE_NAME@
    * gorgone   (systemd:gorgoned):      Started @CENTRAL_SLAVE_NAME@
    * centreon_central_sync     (systemd:centreon-central-sync):         Started @CENTRAL_SLAVE_NAME@
    * cbd_central_broker        (systemd:cbd-sql):       Started @CENTRAL_SLAVE_NAME@
    * centengine        (systemd:centengine):    Started @CENTRAL_SLAVE_NAME@
    * centreontrapd     (systemd:centreontrapd):         Started @CENTRAL_SLAVE_NAME@
    * snmptrapd (systemd:snmptrapd):     Started @CENTRAL_SLAVE_NAME@

Migration Summary:
  * Node: @CENTRAL_MASTER_NAME@:
    * ms_mysql: migration-threshold=1000000 fail-count=1000000 last-failure='Tue Nov  8 17:27:25 2
022'

Failed Resource Actions:
  * ms_mysql_start_0 on @CENTRAL_MASTER_NAME@ 'error' (1): call=440, status='complete', exitreason='M
ariaDB slave io has failed (1236): Got fatal error 1236 from master when reading data from binary
log: 'Error: connecting slave', last-rc-change='Tue Nov  8 17:27:21 2022', queued=0ms, exec=4060ms
```

If you want to switch to the primary node, you must do a failover.

> So, **before to do this, you must check the cluster and database replication status**.

Reviewer: ```suggestion
So, **before you do this, you must check the cluster and database replication status**.
```

> First, check the constraints

Reviewer: ```suggestion
First, check the constraints:
```

```shell
pcs constraint
```

The command should return this:

<Tabs groupId="sync">
<TabItem value="RHEL 8 / Oracle Linux 8 / Alma Linux 8" label="RHEL 8 / Oracle Linux 8 / Alma Linux 8">

```text
Location Constraints:
Ordering Constraints:
  stop centreon then demote ms_mysql-clone (kind:Mandatory)
Colocation Constraints:
  ms_mysql-clone with centreon (score:INFINITY) (rsc-role:Master) (with-rsc-role:Started)
Ticket Constraints:
```

</TabItem>
<TabItem value="RHEL 7 / CentOS 7" label="RHEL 7 / CentOS 7">

```text
Location Constraints:
Ordering Constraints:
  stop centreon then demote ms_mysql-master (kind:Mandatory)
Colocation Constraints:
  ms_mysql-master with centreon (score:INFINITY) (rsc-role:Master) (with-rsc-role:Started)
Ticket Constraints:
```

</TabItem>
</Tabs>

then check the database replication

```bash
/usr/share/centreon-ha/bin/mysql-check-status.sh
```

> At this time, the cluster is on degraded mode whith two slave nodes.

Reviewer: ```suggestion
At this time, the cluster is in degraded mode whith two slave nodes.
```

> At this time, the cluster is in degraded mode whith two slave nodes.

Reviewer: ```suggestion
At this time, the cluster is in degraded mode with two slave nodes.
```

> In this particular case, it return the following information because of ms_mysql resource is stopped on @CENTRAL_MASTER_NAME@:

Reviewer: ```suggestion
In this particular case, it returns the following information because the ms_mysql resource is stopped on @CENTRAL_MASTER_NAME@:
```

```text
Connection SLAVE Status '@CENTRAL_MASTER_NAME@' [KO]
Error reports:
    ERROR 2002 (HY000): Can't connect to MySQL server on '@CENTRAL_MASTER_NAME@' (115)
Impossible de se connecter au serveur '@CENTRAL_MASTER_NAME@'.
Connection SLAVE Status '@CENTRAL_SLAVE_NAME@' [OK]
Slave Thread Status [KO]
Error reports:
    Skip check on '@CENTRAL_MASTER_NAME@'.
    No slave (maybe because we cannot check a server).
Position Status [SKIP]
Error reports:
    Skip because we can't identify a unique slave.
```

You must do a database synchronization from @CENTRAL_SLAVE_NAME@ to @CENTRAL_MASTER_NAME@ by launching the "sync-bigdb" script on the **Slave node**.

```shell
/usr/share/centreon-ha/bin/mysql-sync-bigdb.sh
```

> Like as previous execution of this sript, verify if LVM Snapshot is correctly deleted and Mysql Slave restarted

Reviewer: ```suggestion
As for the previous execution of this script, verify if LVM Snapshot is correctly deleted and the MySQL Slave restarted
```

> As for the previous execution of this script, verify if LVM Snapshot is correctly deleted and the MySQL Slave restarted

Reviewer: ```suggestion
As for the previous execution of this script, verify if LVM Snapshot is correctly deleted and the MySQL Slave restarted:
```

```text
Umount and Delete LVM snapshot
  Logical volume "dbbackupdatadir" successfully removed.
Start MySQL Slave
OK
Start Replication
Id      User    Host    db      Command Time    State   Info    Progress
5       centreon-repl   @CENTRAL_SLAVE_NAME@:51850        NULL    Query   0       starting        show processlist  0.000
6       centreon        localhost       centreon_storage        Sleep   0               NULL    0.000
7       system user             NULL    Connect 0       Connecting to master    NULL    0.000
8       system user             NULL    Slave_SQL       0       Slave has read all relay log; waiting for more updates    NULL    0.000
```

Now, database replication should be OK, verify it.

```shell
/usr/share/centreon-ha/bin/mysql-check-status.sh
```

the result should be:

```text
Connection MASTER Status '@CENTRAL_SLAVE_NAME@' [OK]
Connection SLAVE Status '@CENTRAL_MASTER_NAME@' [OK]
Slave Thread Status [OK]
Position Status [OK]
```

> Now, you can perform a failover to return in initial situation.

Reviewer: ```suggestion
Now, you can perform a failover to return to the initial situation.
```

```shell
pcs resource clear centreon
```

> Do a cleanup to clear errors ans restart ms_mysql resource on @CENTRAL_MASTER_NAME@.

Reviewer: ```suggestion
Do a cleanup to clear errors and restart the ms_mysql resource on @CENTRAL_MASTER_NAME@.
```

```shell
pcs resource cleanup
```

> Situation is stabilized, you can perform a failover by moving centreon resource.

Reviewer: ```suggestion
The situation is stabilized, you can perform a failover by moving the centreon resource.
```

> The situation is stabilized, you can perform a failover by moving the centreon resource.

Reviewer: ```suggestion
The situation is stabilized, you can perform a failover by moving the **centreon** resource.
```

```shell
pcs resource move centreon
```

> The centreon resource is now relocated and cluster is OK, verify with `crm_mon -fr` on any node.

Reviewer: ```suggestion
The centreon resource is now relocated and the cluster is OK, verify with `crm_mon -fr` on any node.
```

> The centreon resource is now relocated and the cluster is OK, verify with `crm_mon -fr` on any node.

Reviewer: ```suggestion
The **centreon** resource is now relocated and the cluster is OK, verify with `crm_mon -fr` on any node.
```
